sites/docs/src/content/docs/guidelines/pipelines/requirements/semantic_versioning.md

```diff
@@ -6,13 +6,22 @@ menu:
     weight: 90
 ---
 
-Pipelines must be released with stable release tags.
-Releases must use GitHub releases and [keep a detailed changelog](https://keepachangelog.com/en/1.0.0/) file.
+- Pipelines must be released with stable release tags.
+- Releases must use GitHub releases and include a [detailed changelog](https://keepachangelog.com/en/1.0.0/) file.
+- Release version tags must be numerical only (no `v` prefix) and should follow [semantic versioning](https://semver.org/) rules: `[major].[minor].[patch]`
 
-Release version tags must be numerical only (no `v` prefix) and should follow [semantic versioning](https://semver.org/) rules: `[major].[minor].[patch]`
+**Versioning Examples**
 
-For example, starting with with a release version `1.4.3`, bumping the version to:
+Starting from release version `1.4.3`, bumping the version to:
 
-- `1.4.4` would be a patch release for minor things such as fixing bugs.
-- `1.5` would be a minor release, for example adding some new features.
-- `2.0` would correspond to the _major_ release where results would no longer be backwards compatible.
+- **`1.4.4`**: A patch release for minor fixes such as bug corrections that do not modify how the user interacts with the pipeline. Examples:
+  - Inserting a missing argument to a process command line
+  - Updating a container build version of a tool that is otherwise the same version of the tool itself
+  - Fixing the logic of the pipeline operators to ensure correct inputs for a process
+- **`1.5.0`**: A minor release that adds new features without changing existing functionality. Examples:
+  - Adding a new parameter that adds new functionality/options to an existing module
+  - Add a new optional or default process that does not change the way the user interacts with previous steps of a pipeline
+- **`2.0.0`**: A major release where execution interaction, inputs or result structures are no longer backwards compatible in terms of pipeline user interaction. Examples:
+  - Renaming or deprecations of existing parameter(s)
+  - Change to input sample sheet specification (adding, dropping, renaming of mandatory columns) or change the input samplesheet specification
+  - Note that a major release DOES NOT necessarily require new functionality
```

Comment:
```suggestion
  - Note that a major release DOES NOT necessarily require new functionality
  - It's expected that regular breaking changes might occur in early stages of pipeline development prior to version 1.0.0, so the criteria don't apply there. 
```

I think we need something like this

Reply:
we discourage to do pre-releases in nf-core, so this is not be needed, I think

Reply:
So the first release should always be V1.0.0? We should probably make that explicit somewhere (maybe not here)

Reply:
I thought we had that somewhere, but don't find it at the moment... let's add it then here and in the "first release" tutorial

Reply:
Done!